fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1825,6 +1880,37 @@ public int reindex(FHIRPersistenceContext context, OperationOutcome.Builder oper
         return result;
     }
 
+    /**
+     * Update the parameters for the resource described by the given DTO
+     * @param <T>
+     * @param rir
+     * @param resourceTypeClass
+     * @param existingResourceDTO
+     * @param reindexDAO
+     * @param operationOutcomeResult
+     * @throws Exception
+     */
+    public <T extends Resource> void updateParameters(ResourceIndexRecord rir, Class<T> resourceTypeClass, com.ibm.fhir.persistence.jdbc.dto.Resource existingResourceDTO,
+        ReindexResourceDAO reindexDAO, OperationOutcome.Builder operationOutcomeResult) throws Exception {
+        if (existingResourceDTO != null && !existingResourceDTO.isDeleted()) {
+            List<String> elements = Collections.emptyList();
+            T existingResource = this.convertResourceDTO(existingResourceDTO, resourceTypeClass, null);
+
+            // Extract parameters from the resource payload we just read and store them, replacing
+            // the existing set
+            reindexDAO.updateParameters(rir.getResourceType(), this.extractSearchParameters(existingResource, existingResourceDTO), rir.getLogicalId(), rir.getLogicalResourceId());
+
+            // Use an OperationOutcome Issue to let the caller know that some work was performed
+            final String diag = "Processed " + rir.getResourceType() + "/" + rir.getLogicalId();
+            operationOutcomeResult.issue(Issue.builder().code(IssueType.INFORMATIONAL).severity(IssueSeverity.INFORMATION).diagnostics(com.ibm.fhir.model.type.String.of(diag)).build());
+        } else {
+            // Reasonable to assume that this resource was deleted because we can't read it
+            final String diag = "Failed to read resource: " + rir.getResourceType() + "/" + rir.getLogicalId();
+            operationOutcomeResult.issue(Issue.builder().code(IssueType.NOT_FOUND).severity(IssueSeverity.WARNING).diagnostics(com.ibm.fhir.model.type.String.of(diag)).build());
```

Comment:
This is where I like to static import the string method. 

import static com.ibm.fhir.model.type.String.string;

string(diag)

makes it short and sweet

Reply:
Yeah, this should be updated. Fixed